azurerm/internal/services/network/express_route_circuit_peering_resource.go

```diff
@@ -320,6 +377,26 @@ func expandExpressRouteCircuitPeeringMicrosoftConfig(input []interface{}) *netwo
 	}
 }
 
+func expandExpressRouteCircuitIpv6PeeringConfig(input []interface{}) *network.Ipv6ExpressRouteCircuitPeeringConfig {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+	peeringConfig := network.Ipv6ExpressRouteCircuitPeeringConfig{
+		PrimaryPeerAddressPrefix:   utils.String(v["primary_peer_address_prefix"].(string)),
+		SecondaryPeerAddressPrefix: utils.String(v["secondary_peer_address_prefix"].(string)),
+		MicrosoftPeeringConfig:     expandExpressRouteCircuitPeeringMicrosoftConfig(v["microsoft_peering_config"].([]interface{})),
+	}
+	routeFilterId := v["route_filter_id"].(string)
```

Comment:
could we add validation to check this is not set when incompatible types are used?

Reply:
I don't know why GitHub does not display the latest code.  I have add a check for `routeFilterId `. https://github.com/xuzhang3/terraform-provider-azurerm/blob/f/expressroutecircuitpeering_enhance/azurerm/internal/services/network/express_route_circuit_peering_resource.go#L400

Reply:
`ipv6`  is a little different with `microsoft_peering_config`, it is only supported by `MicrosoftPeering`.  I add a check for `ipv6`, if peering type is not `MicrosoftPeering` and `ipv6` is configured, an error message will raise up.
Check `ipv6` configure:
https://github.com/xuzhang3/terraform-provider-azurerm/blob/f/expressroutecircuitpeering_enhance/azurerm/internal/services/network/express_route_circuit_peering_resource.go#L267